x-pack/plugins/ml/public/file_datavisualizer/components/file_datavisualizer_view/file_datavisualizer_view.js

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import React, {
+  Component,
+} from 'react';
+
+import {
+  EuiFilePicker,
+  EuiSpacer,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+
+import { ml } from '../../../services/ml_api_service';
+import { ResultsView } from '../results_view';
+import { FileCouldNotBeRead, FileTooLarge } from './file_error_callouts';
+
+
+export class FileDataVisualizerView extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      files: {},
+      fileContents: '',
+      fileSize: 0,
+      fileTooLarge: false,
+      fileCouldNotBeRead: false,
+      loading: false,
+      loaded: false,
+      results: undefined,
+    };
+
+    this.maxPayloadBytes = this.props.maxPayloadBytes;
+  }
+
+  onChange = (files) => {
+    this.setState({
+      files,
+      loading: (files.length > 0),
+      loaded: false,
+      fileContents: '',
+      fileSize: 0,
+      fileTooLarge: false,
+      fileCouldNotBeRead: false,
+      results: undefined,
+    }, () => {
+      if (files.length) {
+        this.analyzeFile(files[0]);
+      }
+    });
+
+  };
+
+  async analyzeFile(file) {
+    if (file.size < this.maxPayloadBytes) {
+      let data = null;
+      try {
+        const fileContents = await readFile(file);
+        data = fileContents.data;
+        this.setState({
+          fileContents: data,
+          fileSize: file.size,
+        });
+
+      } catch (error) {
+        console.error(error);
+      }
+
+      if (data !== null) {
+        try {
+          const resp = await  ml.analyzeFile(data);
+          this.setState({
+            results: resp.results,
+            loaded: true,
+            loading: false,
+            fileTooLarge: false,
+            fileCouldNotBeRead: false,
+          });
+        } catch (error) {
+          console.error(error);
+        }
+      } else {
+        this.setState({
+          loaded: false,
+          loading: false,
+          fileCouldNotBeRead: true,
+          fileSize: file.size,
+        });
+      }
+    } else {
+      this.setState({
+        loaded: false,
+        loading: false,
+        fileTooLarge: true,
+        fileSize: file.size,
+      });
+    }
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <div style={{ textAlign: 'center' }} >
+          <EuiFilePicker
+            id="filePicker"
+            initialPromptText="Select or drag and drop a log file"
```

Comment:
I wonder if this should just say 'Select or drag and drop a file'? Do we need the word 'log' in there?